### PR TITLE
Add x86 ubuntu16.04 image

### PR DESCRIPTION
**Pull request by jashook**
This will enable using the x86_arm crossgen tool in automation.

### CODE REVIEW

**Comment by jashook** [comment]
@MichaelSimons @RussKeldorph @echesakovMSFT PTAL

File: src/ubuntu/14.04/cross/hooks/pre-build [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-$(dirname ${BASH_SOURCE[0]})/../../../build-scripts/build-rootfs.sh ubuntu-14.04 trusty
+$(dirname ${BASH_SOURCE[0]})/../../../../build-scripts/build-rootfs.sh ubuntu-14.04 trusty
```
Comment (MichaelSimons) [inline src/ubuntu/14.04/cross/hooks/pre-build]: Why was this changed?  It doesn't look like the script was moved.

File: src/ubuntu/16.04/x86/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+RUN mkdir -p /home/dotnet-bot
+
+# Set up the dotnet-bot user
+RUN groupadd dotnet-bot && useradd -d /home/dotnet-bot -g dotnet-bot -M -s /bin/bash dotnet-bot
+
+# Set up permissions
+RUN chown -R dotnet-bot:dotnet-bot /home/dotnet-bot
+
+# Setup mount point for coreclr
+RUN mkdir -p /home/dotnet-bot/coreclr
```
Comment (MichaelSimons) [inline src/ubuntu/16.04/x86/Dockerfile]: Why is this necessary?

File: src/ubuntu/16.04/x86/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        gettext \
+        libicu-dev \
+        liblttng-ust-dev \
+        libcurl4-openssl-dev \
+        libssl-dev \
+        uuid-dev \
+        libnuma-dev \
+        libkrb5-dev \
+        binfmt-support \
+    && apt-get clean
```
Comment (MichaelSimons) [inline src/ubuntu/16.04/x86/Dockerfile]: `rm -rf /var/lib/apt/lists/*` is the correct way to clean up the apt-get cache for Docker purposes.

File: src/ubuntu/16.04/x86/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
+FROM i386/ubuntu:16.04
```
Comment (MichaelSimons) [inline src/ubuntu/16.04/x86/Dockerfile]: What is the intended use of this?  This is quite different then the x64 Dockerfile.  I am wondering if this should be tagged more specific to the intended use.

File: src/ubuntu/16.04/x64/cross/hooks/post-build [hunk 5]
```diff
@@ -0,0 +1 @@
+../../../build-scripts/build-rootfs-cleanup.sh
```
Comment (MichaelSimons) [inline src/ubuntu/16.04/x64/cross/hooks/post-build]: Path doesn't seem correct, no ending line feed, no "shebang" - in general this should be consistent with the other post/pre build scripts.

**Comment by jashook** [comment]
Closing for now.
